File: java/spike-scala-servlet/src/main/scala/com/github/gimmi/spikescalaservlet/HelloServlet.scala
package com.github.gimmi.spikescalaservlet

import javax.servlet.http.{HttpServletResponse, HttpServletRequest, HttpServlet}
import com.google.inject.Singleton

@Singleton
class HelloServlet extends HttpServlet {
	override def doGet(req: HttpServletRequest, resp: HttpServletResponse) {
		val encoding = "UTF-8"

		resp setStatus HttpServletResponse.SC_OK
		resp setCharacterEncoding encoding
		resp setContentType s"application/json; charset=$encoding"
		resp.getWriter.write("{ ok: true }")
	}
}

File: java/javase-spa/server/javase-spa/src/main/java/com/sample/javasespa/HelloServlet.java
package com.sample.javasespa;

import java.io.IOException;

import javax.servlet.*;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.*;

// @WebServlet(name = "HelloServlet", urlPatterns = {"/hello"})
public class HelloServlet extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        response.setContentType("text/html");
        response.setStatus(HttpServletResponse.SC_OK);
        response.getWriter().println("<h1>Hello from HelloServlet</h1>");
    }
}
